src/ipa_pc/mod.rs

```diff
@@ -789,89 +919,98 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
                     })?;
-
-                comms.push(commitment);
-                vals.push(*v_i);
+                y_values.push(*y_i);
             }
+            points.push(point);
+        }
 
-            let check_poly = Self::succinct_check(
-                vk,
-                comms.into_iter(),
-                *point,
-                vals.into_iter(),
-                p,
-                opening_challenges,
-            );
+        // Commitment of the h(X) polynomial
+        let batch_commitment = batch_proof.batch_commitment;
 
-            if check_poly.is_none() {
-                return Ok(false);
-            }
+        // Fresh random challenge x
+        let point = Self::compute_random_oracle_challenge(
+            &to_bytes![
+                batch_commitment,
+                points                
+            ]
+            .unwrap(),
+        );
 
-            let check_poly =
-                Polynomial::from_coefficients_vec(check_poly.unwrap().compute_coeffs());
-            combined_check_poly += (randomizer, &check_poly);
-            combined_final_key += &p.final_comm_key.into_projective().mul(&randomizer);
+        let mut opening_challenge_counter = 0;
 
-            randomizer = u128::rand(rng).into();
-            end_timer!(lc_time);
+        // lambda
+        let mut cur_challenge = opening_challenges(opening_challenge_counter);
+        opening_challenge_counter += 1;
+
+        let mut computed_batch_v = G::ScalarField::zero();
+
+        for ((v_i, y_i), x_i) in v_values.clone().into_iter().zip(y_values.clone()).zip(points.clone()) {
+
+            computed_batch_v = computed_batch_v + &(cur_challenge * &((v_i - &y_i) / &(point - &x_i))); 
+
+            cur_challenge = opening_challenges(opening_challenge_counter);
+            opening_challenge_counter += 1;
         }
 
-        let proof_time = start_timer!(|| "Checking batched proof");
-        let final_key = Self::cm_commit(
-            vk.comm_key.as_slice(),
-            combined_check_poly.coeffs.as_slice(),
-            None,
-            None,
+        v_values.push(computed_batch_v);
+
+        let mut commitments = commitments.clone().into_iter().collect::<Vec<&'a LabeledCommitment<Self::Commitment>>>();
+        let labeled_batch_commitment = LabeledCommitment::new(
+            format!("Batch"),
+            Commitment { comm: batch_commitment, shifted_comm: None },
+            None
         );
-        if !ProjectiveCurve::is_zero(&(final_key - &combined_final_key)) {
-            return Ok(false);
-        }
+        commitments.push(&labeled_batch_commitment);
 
-        end_timer!(proof_time);
+        end_timer!(batch_check_time);
 
-        Ok(true)
+        let proof = &batch_proof.proof;
+
+        Self::check_individual_opening_challenges(
+            vk,
+            commitments,
+            point,
+            v_values,
+            proof,
+            opening_challenges,
```

Comment:
Same as above: this should be a new challenge

Reply:
Done